iotdb-core/consensus/src/main/java/org/apache/iotdb/consensus/iot/service/IoTConsensusRPCServiceProcessor.java

```diff
@@ -244,7 +244,7 @@ public void waitSyncLogComplete(
       return;
     }
     long searchIndex = impl.getSearchIndex();
-    long safeIndex = impl.getCurrentSafelyDeletedSearchIndex();
+    long safeIndex = impl.getMinFlushedIndex();
```

Comment:
should not use flushedIndex here because the index maybe not be flushed

Reply:
Done.